packages/node/src/ethereum/ethers/json-rpc-batch-provider.ts

```diff
@@ -146,6 +174,28 @@ export class JsonRpcBatchProvider extends JsonRpcProvider {
         batch.forEach((inflightRequest) => {
           inflightRequest.reject(error);
         });
+
+        //this.adjustBatchSize(false);
       });
   }
+
+  private adjustBatchSize(success: boolean) {
+    success ? this.successfulBatchCount++ : this.failedBatchCount++;
+    const totalBatches = this.successfulBatchCount + this.failedBatchCount;
+
+    if (totalBatches % this.batchSizeAdjustmentInterval === 0) {
+      const successRate = this.successfulBatchCount / totalBatches;
+
+      // Adjust the batch size based on the success rate.
+      if (successRate < 0.9 && this.batchSize > 1) {
+        this.batchSize--;
```

Comment:
Do we have to limit the batch size to make sure it stays above 0

Reply:
`if (successRate < 0.9 && this.batchSize > 1)` 

We have a check to make sure it stays above 0